base/linalg/arpack.jl

```diff
@@ -122,9 +122,9 @@ function eupd_wrapper(T, n::Integer, sym::Bool, cmplx::Bool, bmat::String,
     elseif which == "SR" || which == "SA"
         dmap = x->-real(x)
     elseif which == "LI"
-        dmap = imag
+        dmap = x -> imag.(x)
     elseif which == "SI"
-        dmap = x->-imag(x)
+        dmap = x -> (-).(imag.(x))
```

Comment:
That looks like a winking owl. The way you have it here gets rid of the real component before negating. Not sure if the compiler would elide negating the real component if you did it the other way around.